flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/ProcessingTimeAware.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.tasks;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * Interface for classes that needs access to the {@link ProcessingTimeService}
+ * to get the current processing time, register timers, etc.
+ */
+@PublicEvolving
+public interface ProcessingTimeAware {
```

Comment:
To me `ProcessingTimeServiceAware` would be a better name, as it's more precise.